packages/ember-intl/addon/macros.js

```diff
@@ -0,0 +1,57 @@
+import Ember from 'ember';
+
+const { get, assert, computed } = Ember;
+const keys = Object.keys;
+
+function extractDependencies(obj) {
+  return keys(obj)
+    .map(key => obj[key])
+    .filter(value => value.isReference)
+    .map(it => it.key);
+}
+
+function resolve(context, value) {
+  return value.isReference ? get(context, value.key) : value;
+}
+
+function resolveHash(context, hash) {
+  let result = {};
+
+  keys(hash).forEach(function(key) {
+    result[key] = resolve(context, hash[key]);
+  });
+
+  return result;
+}
+
+export function raw(value) {
+  return { value, isRaw: true };
+}
+
+export function ref(key) {
```

Comment:
I think we should have only `raw`.  Not using a wrapper, when providing a string, should default to `ref`.  != String should default to `raw`.